File: src/main/java/io/awilson/timein/services/InstructorServiceImpl.java
package io.awilson.timein.services;

import io.awilson.timein.domain.Instructor;
import io.awilson.timein.repositories.InstructorRepository;
import lombok.Setter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class InstructorServiceImpl implements InstructorService {

    @Setter(onMethod = @_(@Autowired))
    private InstructorRepository instructorRepository;

    @Override
    public Iterable<Instructor> listAllInstructors() {
        return instructorRepository.findAll();
    }

    @Override
    public Instructor getInstructorById(Integer id) {
        return instructorRepository.findOne(id);
    }

    @Override
    public Instructor saveInstructor(Instructor instructor) {
        return instructorRepository.save(instructor);
    }

    @Override
    public void deleteInstructor(Integer id) {
        instructorRepository.delete(id);
    }
}

File: src/main/java/io/awilson/timein/services/InstructorServiceImpl.kt
package io.awilson.timein.services

import io.awilson.timein.domain.Instructor
import io.awilson.timein.repositories.InstructorRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class InstructorServiceImpl : InstructorService {

    @Autowired
    lateinit var repo: InstructorRepository

    override fun listAllInstructors(): Iterable<Instructor> = repo.findAll()

    override fun getInstructorById(id: Int): Instructor = repo.findOne(id)
            ?: throw Exception("Instructor:$id not found")

    override fun saveInstructor(instructor: Instructor): Instructor = repo.save(instructor)

    override fun deleteInstructor(id: Int) = repo.delete(id)
}